### PR TITLE
Update ServiceImport aggregation proposal

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

File: lighthouse/aggregate-service-import.md [hunk 1]
```diff
@@ -18,18 +18,15 @@ added to the Lighthouse agent:
...
-  if needed. The service VIPs from the exporting clusters should be stored in `EndpointSlices`,
-  with a separate `EndpointSlice` per exporting cluster.
-
-- The current GlobalNet implementation exports `EndpointSlices` for normal services that contain
-  pod IP addresses from the originating cluster. The `EndpointSlice` purpose will need to change
-  to contain the Globalnet service VIP for the originating cluster.
+Currently, each cluster exporting a service creates a `ServiceImport` containing its local service VIP and service ports.
+However the individual cluster information cannot be aggregated into a single `ServiceImport`. The MCS specification
+intends for cluster-specific information to be stored in `EndpointSlices`, with a separate `EndpointSlice` per exporting
+cluster. Currently, lighthouse does export `EndpointSlices` per cluster containing the local endpoints that is simply
```
Comment (skitt) [inline lighthouse/aggregate-service-import.md]:
```suggestion
cluster. Currently, Lighthouse does export `EndpointSlices` per cluster containing the local endpoints that are simply
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: lighthouse/aggregate-service-import.md [hunk 2]
```diff
@@ -83,55 +93,53 @@ than is possible with the DNS resolver.
...
-1. If not in conflict, lighthouse updates `ServiceImport.Status.Clusters` to include the cluster
-   name.
-1. A new `EndpointSlice` is created for the service VIP (or Pod VIPs for headless) and synced to
+1. When a new `ServiceExport` is created, a local `ServiceImport` is created containing the local `Service`
+   information.
+2. The local `ServiceImport` is checked for conflicts with the aggregated `ServiceImport` located on the broker.
+3. If in conflict, a `ServiceExportConflict` condition is set in the `ServiceExport` status and conflict resolution is
+   attempted.
+4. If a conflict cannot be resolved, the cluster's service information is not exported.
+5. Otherwise or if no conflicts, Lighthouse updates `ServiceImport.Status.Clusters` to include the cluster
```
Comment (aswinsuryan) [inline lighthouse/aggregate-service-import.md:103]: Will we continue to have ServiceImport in the broker namespace of the local cluster? I was not clear about that.

Comment (tpantelis) [reply lighthouse/aggregate-service-import.md:103]: Yes it will be aggregated in the broker namespace of the broker cluster (local or not).

Comment (tpantelis) [reply lighthouse/aggregate-service-import.md:103]: I updated the section to reflect this.

**Review by aswinsuryan — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Review by vthapar — APPROVED** [review]

**Pull request merged**